### PR TITLE
Unify all storage interface behavior

### PR DESCRIPTION
**Pull request by critical27**
Modify according to https://github.com/vesoft-inc/nebula-common/pull/323.

Another involved change is the vertex cache. We only add to cache when read, and evict in all other operations.

### CODE REVIEW

File: src/storage/exec/IndexOutputNode.h [hunk 1]
```diff
@@ -219,27 +218,26 @@ class IndexOutputNode final : public RelNode<T> {
...
+                    row.emplace_back(planContext_->edgeType_);
+                } else if (colName == kRank) {
+                    row.emplace_back(NebulaKeyUtils::getRank(planContext_->vIdLen_, val.first));
+                } else if (colName == kDst) {
+                    auto dst = NebulaKeyUtils::getDstId(planContext_->vIdLen_, val.first);
+                    addVidToRow(dst, row);
+                } else {
+                    auto v = reader->getValueByName(colName);
+                    row.emplace_back(std::move(v));
+                }
```
Comment (bright-starry-sky) [inline src/storage/exec/IndexOutputNode.h:240]: Why need to collect TagId and EdgeType in index result set?  I guess the tagId and EdgeType are used for pipe . right?

File: src/storage/index/LookupBaseProcessor.inl [hunk 2]
```diff
@@ -123,6 +92,14 @@ bool LookupBaseProcessor<REQ, RESP>::isOutsideIndex(Expression* filter,
...
             break;
         }
+        case Expression::Kind::kEdgeSrc:
+        case Expression::Kind::kEdgeType:
+        case Expression::Kind::kEdgeRank:
+        case Expression::Kind::kEdgeDst: {
+            auto* sExpr = static_cast<PropertyExpression*>(filter);
+            auto propName = *(sExpr->prop());
+            return propsInEdgeKey.find(propName) == propsInEdgeKey.end();
+        }
```
Comment (bright-starry-sky) [inline src/storage/index/LookupBaseProcessor.inl:102]: What's the purpose here? I think here need  log `FATAL` or `ERROR`.

File: src/storage/index/LookupBaseProcessor.inl [hunk 3]
```diff
@@ -161,106 +138,101 @@ StatusOr<StoragePlan<IndexID>> LookupBaseProcessor<REQ, RESP>::buildPlan() {
...
-                auto filter = Expression::decode(ctx.get_filter());
-                auto isFieldsOutsideIndex = isOutsideIndex(filter.get(), indexItem);
-                if (isFieldsOutsideIndex) {
-                    needData = needFilter = true;
-                }
+            auto it = std::find_if(fields.begin(),
+                                    fields.end(),
+                                    [&yieldCol] (const auto& columnDef) {
+                                        return yieldCol == columnDef.get_name();
+                                    });
```
Comment (bright-starry-sky) [inline src/storage/index/LookupBaseProcessor.inl:178]: alignment

File: src/storage/index/LookupProcessor.cpp [hunk 4]
```diff
@@ -43,6 +43,17 @@ void LookupProcessor::process(const cpp2::LookupIndexRequest& req) {
...
+        std::transform(resultDataSet_.colNames.begin(),
+                       resultDataSet_.colNames.end(),
+                       resultDataSet_.colNames.begin(),
+                       [this](const auto& col) { return planContext_->edgeName_ + "." + col; });
+    } else {
+        std::transform(resultDataSet_.colNames.begin(),
+                       resultDataSet_.colNames.end(),
+                       resultDataSet_.colNames.begin(),
+                       [this](const auto& col) { return planContext_->tagName_ + "." + col; });
+    }
```
Comment (bright-starry-sky) [inline src/storage/index/LookupProcessor.cpp:56]: Excellent. convert when you're done.

File: src/storage/exec/IndexOutputNode.h (hunk 1 above)
Comment (critical27) [reply src/storage/exec/IndexOutputNode.h:240]: Yeah, usually we don't need it, if query engine specify it, need to return it as well..

File: src/storage/index/LookupBaseProcessor.inl (hunk 2 above)
Comment (critical27) [reply src/storage/index/LookupBaseProcessor.inl:102]: These expression could convert to `PropertyExpression`, we could support filter like `prefix(dst) == "abc"` later.

File: src/storage/index/LookupBaseProcessor.inl (hunk 3 above)
Comment (critical27) [reply src/storage/index/LookupBaseProcessor.inl:178]: good catch

**Review by bright-starry-sky — DISMISSED** [review]
LGTM

**Comment by bright-starry-sky** [comment]
LGTM

**Review by bright-starry-sky — APPROVED** [review]

**Review by panda-sheep — APPROVED** [review]
Well done，Excellent!

**Review by liuyu85cn — APPROVED** [review]
LGTM

**Pull request merged**